src/main.rs

```diff
@@ -7,35 +7,56 @@ use nalgebra;
 mod rendering;
 
 use std::io::BufRead;
+use std::time::Instant;
 use winit::{
     event::{Event, WindowEvent},
     event_loop::{ControlFlow, EventLoop},
     window::WindowBuilder,
 };
 
+use imgui::*;
+use imgui_winit_support;
+
 enum MouseMode {
     Cursor,
     CameraLook,
     CameraPan,
 }
 
+struct UiState {
+    camera_target: [f32; 3],
+    camera_range: f32,
+    camera_azimuth: f32,
+    camera_elevation: f32,
+    gui_open: bool,
+}
+
+impl UiState {
+    pub fn default() -> Self {
+        return UiState {
+            camera_target: [0.0, 0.0, 0.0],
+            camera_range: 10.0,
+            camera_azimuth: 0.0,
+            camera_elevation: 0.0,
+            gui_open: false,
+        };
+    }
+}
```

Comment:
When you move the UI processing out into it's own thing you should move this with it.